imblearn/datasets/zenodo.py

```diff
@@ -0,0 +1,281 @@
+from sklearn.datasets.base import Bunch
+from sklearn.utils.fixes import makedirs
+from sklearn.externals import six
+from sklearn.utils import check_random_state
+
+URL = ('https://zenodo.org/record/61452/files/'
+       'benchmark-imbalanced-learn.tar.gz')
+PRE_FILENAME = 'x'
+POST_FILENAME = 'data.npz'
+
+MAP_NAME_ID_KEYS = ['ecoli',
+                    'optical_digits',
+                    'satimage',
+                    'pen_digits',
+                    'abalone',
+                    'sick_euthyroid',
+                    'spectrometer',
+                    'car_eval_34',
+                    'isolet',
+                    'us_crime',
+                    'yeast_ml8',
+                    'scene',
+                    'libras_move',
+                    'thyroid_sick',
+                    'coil_2000',
+                    'arrhythmia',
+                    'solar_flare_m0',
+                    'oil',
+                    'car_eval_4',
+                    'wine_quality',
+                    'letter_img',
+                    'yeast_me2',
+                    'webpage',
+                    'ozone_level',
+                    'mammography',
+                    'protein_homo',
+                    'abalone_19']
+
+MAP_NAME_ID = OrderedDict()
+MAP_ID_NAME = OrderedDict()
+for k, v in zip(MAP_NAME_ID_KEYS, range(1, 28)):
+    MAP_NAME_ID[k] = v
+    MAP_ID_NAME[v] = k
+
+logger = logging.getLogger()
+
+
+def fetch_zenodo(data_home=None,
+                 filter_data=None,
+                 download_if_missing=True,
+                 random_state=None,
+                 shuffle=False):
+    """Load the Higgs dataset, downloading it if necessary.
+
+    Read more in the :ref:`User Guide <datasets>`.
+
+    Parameters
+    ----------
+    data_home : string, optional (default=None)
+        Specify another download and cache folder for the datasets. By default
+        all scikit learn data is stored in '~/scikit_learn_data' subfolders.
+
+    filter_data : tuple of str/int or None, optional (default=None)
+        A tuple containing the ID or the name of the datasets to be returned.
+        Refer to the above table to get the ID and name of the datasets.
+
+    download_if_missing : boolean, optional (default=True)
+        If False, raise a IOError if the data is not locally available
+        instead of trying to download the data from the source site.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        Random state for shuffling the dataset.
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+    shuffle : bool, optional (default=False)
+        Whether to shuffle dataset.
+
+    Returns
+    -------
+    datasets : OrderedDict of Bunch object,
```

Comment:
I would like to specified that this is an ordered dictionary. I change the code in accordance